config/module/tree.go

```diff
@@ -297,9 +301,145 @@ func (t *Tree) Load(s getter.Storage, mode GetMode) error {
 	// Set our tree up
 	t.children = children
 
+	// if we're the root module, we can now set the provider inheritance
+	if len(t.path) == 0 {
+		t.inheritProviderConfigs(nil)
+	}
+
 	return nil
 }
 
+// Once the tree is loaded, we can resolve all provider config inheritance.
+//
+// This moves the full responsibility of inheritance to the config loader,
+// simplifying locating provider configuration during graph evaluation.
+// The algorithm is much simpler now too. If there is a provider block without
+// a config, we look in the parent's Module block for a provider, and fetch
+// that provider's configuration. If that doesn't exist, we assume a default
+// empty config. Implicit providers can still inherit their config all the way
+// up from the root, so we walk up the tree and copy the first matching
+// provider into the module.
```

Comment:
This is totally a minor thing, but it feels weird to me to have comment text talking about a change, since this comment will probably be around long after we've forgotten there was inheritance handling in the `terraform` package.

Certainly not the most important thing, but if possible it'd be nice to try to reword this so that it's written more like a description of the current behavior in isolation, rather than comparing it to a previous state.

(These are good words for a commit message, though!)

Reply:
Agree. I'll re-word this strictly as method documentation.